c++/src/Reader.cc

```diff
@@ -385,7 +385,7 @@ namespace orc {
 
   void RowReaderImpl::seekToRowGroup(uint32_t rowGroupEntryId) {
     // store positions for selected columns
-    std::vector<std::list<uint64_t>> positions;
+    std::list<std::list<uint64_t>> positions;
```

Comment:
Could you elaborate a little more about your PR description?
> Reallocation of positions vector may invalidate contained iterators.

Reply:
In my environment (Windows, MSVC 19.12.25835.0), 
test DictionaryEncoding.multipleStripesWithIndex failed due to invalid iterators in positionProviders.

      // copy index positions for a specific column
      positions.push_back({});
      auto& position = positions.back();
      for (int pos = 0; pos != entry.positions_size(); ++pos) {
        position.push_back(entry.positions(pos));             // iterators can be invalidated here due to reallocation
      }
      positionProviders.insert(std::make_pair(colId, PositionProvider(position))); //    iterator taken here

Reply:
Thank you for the details. Could you add that into the PR description? It will be a permanent commit message.

Reply:
Just a question: We have a Window test coverage with AppVeyor(Windows, MSVC 19.0.24241.7) which is healthy. Do you think the failure is due to the behavior change in somewhere of MSVC 19.12.x?
- https://ci.appveyor.com/project/ApacheSoftwareFoundation/orc/builds/40680172

Reply:
according to the specs (https://en.cppreference.com/w/cpp/container#Iterator_invalidation)
vector iterators are invalid if insertion changed capacity.
May be initial capacity of vector changed in new version msvc.
You probably can reproduce this issue using the following patch:
`
--- a/c++/src/Reader.cc
+++ b/c++/src/Reader.cc
@@ -401,6 +401,7 @@ namespace orc {
       for (int pos = 0; pos != entry.positions_size(); ++pos) {
         position.push_back(entry.positions(pos));
       }
+      auto size = position.size(); position.resize(100500); position.resize(size);
       positionProviders.insert(std::make_pair(colId, PositionProvider(position)));
     }
`